tools/fastq_provider/fastq_provider.xml

```diff
@@ -1,39 +1,77 @@
+            #if $download_type:
+            -d '$download_type'
+            #end if
+        #else:
+        fetchEnaLibraryFastqs.sh -l '$library' -d ./
+            #if $sepe:
+            -n '$sepe'
+            #end if
+            #if $download_type:
+            -t '$download_type'
+            #end if
+        #end if
         #if $resource
         -s '$resource' 
         #end if
         #if $retrieval_method
         -m '$retrieval_method'
         #end if
-        #if $mode
-        -p '$mode'
-        #end if
         #if $library
         -l '$library'
         #end if
         #if $config_file
         -c '$config_file'
         #end if
-        #if $validate_only
-        -v '$validate_only'
-        #end if
-        #if $download_type
-        -d '$download_type' 
+        #if $mode
+        -p '$mode'
         #end if
+        #if $main_source == 'file_or_uri':
         ; mv -f '$target' '$outfile' 
+        #end if
     ]]></command> 
     <inputs>
-        <param name="source" argument="-f" type="text" label="file or uri" help="file or uri" />
-        <param name="target" argument="-t" type="text"  label="target file" help="target file. E.g 'ERR035229.fastq.gz' " />
+
+        <conditional name="fetchFastq">
+            <param name="main_source" argument="-a" type="select" label="file_or_uri or library" help="file_or_uri or library" >
+                <option value="library" selected="true">public</option>
+                <option value="file_or_uri">private</option>
+            </param>
+            <when value="file_or_uri">
+                <param name="source" argument="-f" type="text" label="file or uri" help="file or uri" />
+                <param name="target" argument="-t" type="text"  label="target file" help="target file. E.g 'ERR035229.fastq.gz' " />
+                <param name="validate_only" argument="-v" type="boolean" label="validate only to check a source URI is valid, don't download" help="validate only, don't download" optional='true'/>
+                <param name="download_type" argument="-d" type="select" label="download type, fastq or SRA file" help="download type, fastq or srr" optional='true'>
+                    <option value="fastq" selected="true">fastq</option>
+                    <option value="srr">SRA</option>
+                </param>
+            </when>
+            <when value="library">
+                <param name="library" argument="-l" type="text" label="library" help="library" />
+                <param name="download_type" argument="-t" type="select" label="download type, fastq or SRA file" help="download type, fastq or srr" optional='true'>
+                    <option value="fastq" selected="true">fastq</option>
+                    <option value="srr">SRA</option>
+                </param>
+                <param name="sepe" argument="-n" type="select" label="SINGLE-end or PAIRED-end, default PAIRED" help="SINGLE-end or PAIRED-end, default PAIRED" optional='true'>
```

Comment:
Can't we get paired end files when requesting through file_or_uri? I would expect this option to be outside fo the fetchFastq conditional

Reply:
According to the [tool readme](https://github.com/ebi-gene-expression-group/atlas-fastq-provider), the downloading by file or URI using fetchFastq.sh is indeed just for individual files.